### PR TITLE
ci-wrapper: grep for errors at the end

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by SkySkimmer** [comment]
added color support

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by SkySkimmer** [comment]
@ejgallego ping

**Comment by ejgallego** [comment]
@SkySkimmer did you close on purpose?

This LGTM, for CI projects that use Dune, we already can enable a feature like this one.

**Comment by ejgallego** [comment]
@coqbot run full CI

Let's run this "just in case", then we can merge.

**Comment by SkySkimmer** [comment]
We can see that it works thanks to some unrelated failures.

**Comment by SkySkimmer** [comment]
ping @ejgallego

File: dev/ci/ci-wrapper.sh [hunk 1]
```diff
@@ -38,5 +38,49 @@ else
...
+    | tr -d '\0' > errors \
+    || true
+
+  if [ -s errors ]; then {
+    echo
+    echo "Error list (may be incomplete):"
+    echo
+    cat errors
+  } >&2
+  fi
```
Comment (ejgallego) [inline dev/ci/ci-wrapper.sh:82]: maybe print on `else` too?
```sh
else {
   echo "Error list is empty (but program exited with error code)"
}
```

**Review by ejgallego — APPROVED** [review]
I'm always a bit uneasy about grepping for stuff that is often free-from, but this seems OK to me.

For Coq itself, we could use `--error-reporting=deterministic` in the CI.

File: dev/ci/ci-wrapper.sh (hunk 1 above)
Comment (SkySkimmer) [reply dev/ci/ci-wrapper.sh:82]: I don't think we need to add extra prints.

**Comment by ejgallego** [comment]
> > For Coq itself, we could use --error-reporting=deterministic in the CI.
> 
> `man dune build` says
> 
> ```
>        --error-reporting=VAL (absent=deterministic)
> ```
> 
> is it not the default?

Interesting, I have `(absent=early)` myself. Did you tweak your `~/.config/dune/config` file?

**Comment by SkySkimmer** [comment]
I don't think the config file changes what man says.

**Comment by ejgallego** [comment]
Maybe it is dune to `INSIDE_DUNE`?

Code in `report_errors_config.ml`:
```ocaml
let default : t =
  match Dune_util.Execution_env.inside_dune with
  | true -> Deterministic
  | false -> Early
```

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**